go/multipooler/manager/manager.go

```diff
@@ -394,6 +394,36 @@ func (pm *MultiPoolerManager) getPoolerType() clustermetadatapb.PoolerType {
 	return clustermetadatapb.PoolerType_UNKNOWN
 }
 
+// getDatabase returns the database name from the multipooler record
+func (pm *MultiPoolerManager) getDatabase() string {
+	pm.cachedMultipooler.mu.Lock()
+	defer pm.cachedMultipooler.mu.Unlock()
+	if pm.cachedMultipooler.multipooler != nil && pm.cachedMultipooler.multipooler.MultiPooler != nil {
+		return pm.cachedMultipooler.multipooler.Database
+	}
+	return ""
+}
+
+// getBackupLocation returns the backup location from the database topology
+func (pm *MultiPoolerManager) getBackupLocation(ctx context.Context) (string, error) {
+	database := pm.getDatabase()
```

Comment:
One idea: I wonder if we should read the DB information during startup and keep it in MultiPoolerManager. I don't expect the DB and backup location to change once the process starts, so we can avoid hitting topo every time we need it.